flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -254,55 +272,89 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
   }
 
   Future<void> _finishTimeToDisplayTracking() async {
-    _timeToDisplayTracker?.clear();
+    try {
+      final transaction = _transaction;
+      _transaction = null;
+      if (transaction == null || transaction.finished) {
+        return;
```

Comment:
we are returning too early imho
if a transaction is finished because it was in manual mode, we don't clear the tracker, and it will expect the next transaction to be manual, right?

Reply:
it's wrapped in a try / finally with `_clear()` and when we return the finally block is executed as well

Reply:
sry now I know what you mean, yes I'll fix that

Reply:
clear now is only called if `clearAfter` is true (in `didPop`) and additionally it is also cleared at the very start of `didPush` so we can make sure we start from a clean state. It's safe to call because didPush happens before the Widget is being built so we wouldn't override any manual flags